platform/api/esx/api.go

```diff
@@ -383,11 +383,13 @@ func (a *API) CreateDevice(name string, conf *conf.Conf, ips *IpPair) (*ESXMachi
 		plog.Debugf("Configuring userdata %q", userdata)
 		err = a.updateGuestVariable(vm, "guestinfo.ignition.config.data", userdata)
 		if err != nil {
+			_ = a.deleteDevice(vm)
```

Comment:
Instead of repeating this call every time, I think it would be better to use a defer right after creating the VM.
I think this is would you'd need:
```
	defer func() {
		if err != nil {
			_ = a.deleteDevice(vm)
		}
	}()
```

Reply:
Definitely but that only works if `err` is not shadowed by accident. Added a comment on this and converted a case which is not strictly shadowing `err` but would be if it is moved into a bracket later.

Reply:
Alright, looks good!